### PR TITLE
Add overall same ratings

### PR DESCRIPTION
**Pull request by maxyuan6717**

### CODE REVIEW

File: api/queries/catalog.queries.js [hunk 1]
```diff
@@ -21,6 +21,10 @@ export const catalogBySeasonQuery = gql`
       average_professor
       average_rating
       average_workload
+      course {
+        average_rating_same_professors
+        average_workload_same_professors
+      }
```
Comment (hsheth2) [inline api/queries/catalog.queries.js:27]: @kevinhu can we get these copied into the generated table

File: frontend/src/components/SearchResultsGridItem.js [hunk 2]
```diff
@@ -290,17 +298,12 @@ const SearchResultsGridItem = ({
...
                     style={{
-                      color:
-                        course.average_rating && isLoggedIn
-                          ? ratingColormap(course.average_rating)
-                              .darken()
-                              .saturate()
-                          : '#cccccc',
+                      color: course_rating
+                        ? ratingColormap(course_rating).darken().saturate()
+                        : '#cccccc',
```
Comment (hsheth2) [inline frontend/src/components/SearchResultsGridItem.js:303]: Not sure if this is related but the `na_cell` field in constants.js is unused

File: frontend/src/utilities.js [hunk 3]
```diff
@@ -269,3 +269,21 @@ export const sortCourses = (courses, ordering) => {
...
+    ? course['course.average_rating_same_professors'].toFixed(1) // Use same professor if possible
+    : course.average_rating
+    ? course.average_rating.toFixed(1) // Use all professors otherwise
+    : null; // No ratings at all
+  // String representation of rating to be displayed
+  const course_rating_str = course['course.average_rating_same_professors']
+    ? course_rating // Use same professor if possible. Displayed as is
+    : course.average_rating
+    ? `~${course_rating}` // Use all professors otherwise and add tilda ~
+    : 'N/A'; // No ratings at all
```
Comment (hsheth2) [inline frontend/src/utilities.js:285]: this entire thing seems overly complex - why do we need to return `course_rating_str` here?

File: api/queries/catalog.queries.js (hunk 1 above)
Comment (kevinhu) [reply api/queries/catalog.queries.js:27]: Done! FYI @maxyuan6717 we now have columns indicating the number of classes used to calculate ratings (just the name with `_n` at the end).

**Review by kevinhu — APPROVED** [review]

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**